variables.tf

```diff
@@ -231,3 +231,9 @@ variable "retention" {
     error_message = "Variable must not be null."
   }
 }
+
+variable "enable_service_monitor" {
+  description = "Enable Prometheus ServiceMonitor in the Helm chart."
+  type        = bool
+  default     = true
+}
```

Comment:
Could you please put this variable between the `target_revision` and `helm_values` variables inside the `Standard Variables` section ([example 1](https://github.com/camptocamp/devops-stack-module-traefik/blob/b4d0db1b0bfe57f3b2c766a42153e2ae3bc52fdc/variables.tf#L29))? I think it's good practice to organize the variables the same way throughout the modules.

On another note, you are missing something important: this variable will not work unless you propagate these calls from the variants until the root module. That is, when each variant calls the root module, they need to pass the same variable ([example 2](https://github.com/camptocamp/devops-stack-module-traefik/blob/b4d0db1b0bfe57f3b2c766a42153e2ae3bc52fdc/eks/main.tf#L8)).